core/src/main/java/com/digitalpebble/stormcrawler/protocol/file/FileResponse.java

```diff
@@ -0,0 +1,193 @@
+            LOG.warn("url.getPath() != url.getFile(): {}.", url);
+        }
+
+        String path = "".equals(url.getPath()) ? "/" : url.getPath();
+
+        File file = new File(
+                URLDecoder.decode(path, fileProtocol.getEncoding()));
+
+        if (!file.exists()) {
+            statusCode = HttpStatus.SC_NOT_FOUND;
+            return;
+        }
+
+        if (!file.canRead()) {
+            statusCode = HttpStatus.SC_UNAUTHORIZED;
+            return;
+        }
+
+        if (!file.equals(file.getCanonicalFile())) {
+            metadata.setValue(HttpHeaders.LOCATION, file.getCanonicalFile()
+                    .toURI().toURL().toString());
+            statusCode = HttpStatus.SC_MULTIPLE_CHOICES;
+            return;
+        }
+
+        if (file.isDirectory()) {
+            getDirAsHttpResponse(file);
+        } else if (file.isFile()) {
+            getFileAsHttpResponse(file);
+        } else {
+            statusCode = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+            return;
+        }
+
+        if (content == null) {
+            content = new byte[0];
+        }
+    }
+
+    public ProtocolResponse toProtocolResponse() throws IOException {
+        return new ProtocolResponse(content, statusCode, metadata);
+    }
+
+    private void getFileAsHttpResponse(File file) throws IOException {
+        long size = file.length();
+
+        if (size > Integer.MAX_VALUE) {
+            statusCode = HttpStatus.SC_BAD_REQUEST;
+            return;
+        }
+
+        int len = (int) size;
+
+        if (fileProtocol.getMaxContentLength() >= 0
+                && len > fileProtocol.getMaxContentLength())
+            len = fileProtocol.getMaxContentLength();
+
+        content = new byte[len];
+
+        try (InputStream is = new FileInputStream(file)) {
+            int offset = 0;
+            int n = 0;
+            while (offset < len
+                    && (n = is.read(content, offset, len - offset)) >= 0) {
+                offset += n;
+            }
+
+            if (offset < len) {
+                LOG.warn("not enough bytes read from file: {}", file.getPath());
+            }
+        } catch (IOException e) {
+            String errorMessage = "Exception while fetching file response "
+                    + file.getPath() + ": " + e;
+            LOG.error(errorMessage, e);
+            statusCode = HttpStatus.SC_METHOD_FAILURE;
+            return;
+        }
+
+        metadata.setValue(HttpHeaders.CONTENT_LENGTH, new Long(size).toString());
+        metadata.setValue(HttpHeaders.LAST_MODIFIED,
+                formatDate(file.lastModified()));
+
+        statusCode = HttpStatus.SC_OK;
+    }
+
+    private void getDirAsHttpResponse(File file) {
+        content = generateSitemap(file);
+        metadata.setValue(HttpHeaders.CONTENT_TYPE, "application/xml");
+        metadata.setValue("isSitemap", "true");
+        statusCode = HttpStatus.SC_OK;
+    }
+
+    private String formatDate(long date) {

```

Comment:
Mostly because the dateFormat itself is static and there is nothing specific to the instance. It's just a utility function which would return the same result regardless of the fields of the instance. Does that answer your question?